src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1462,6 +1467,13 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
     private AuthenticationResult localFlow(CallbackHandler callbackHandle,
             final IWindowComponent activity, final boolean useDialog,
             final AuthenticationRequest request) {
+        // Update the PromptBehavior. Since we add the new prompt behavior(force_prompt) for broker apps to 
+        // force prompt, if this flag is set in the embeded flow, we need to update it to always. For embed 
+        // flow, force_prompt is the same as always. 
+        if (PromptBehavior.FORCE_PROMPT == request.getPrompt()) {
```

Comment:
@weijjia I think it would be better to handle the equivalence of prompt behaviors in the place where it is interpreted.  In other words, somewhere later the code is making a decision about whether to prompt or not based on this enum.  That is the right place to make these flags equivalent.  Having this here seems a bit random.